### PR TITLE
Internal: add visual regression tests

### PR DESCRIPTION
**Pull request by christianvuerings**
### Summary

Adds visual regression tests to Gestalt. We added tests for 2 components: `Button` and `Text`, both light and dark mode tests.

#### Notes

* We always give the screenshot a specific name. That way, we can use it in our docs
* We use `display="inlineBlock"` to only capture the component (not the whole page)
* Uses a Cypress plug-in called [cypress-plugin-snapshots](https://github.com/meinaart/cypress-plugin-snapshots)

### Example to show how to update tests

https://user-images.githubusercontent.com/127199/132586626-c5bf973b-070d-47a8-b329-930124e34e9d.mp4

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -99,21 +100,21 @@
   "scripts": {
     "a11y:generate": "./scripts/generateAccessibilitySpecs.js",
     "a11y:validate": "./scripts/validateAccessibilitySpecs.js",
-    "svgIcons:validate": "./scripts/validateIconSvgSpecs.js",
```
Comment (christianvuerings) [inline package.json:102]: alphabetized these

File: cypress/integration/visual-test/Button-dark_spec.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+describe('Button dark mode visual regression check', () => {
```
Comment (dangerismycat) [inline cypress/integration/visual-test/Button-dark_spec.js:1]: Can we combine the dark and light mode tests into one file? I'm thinking that could reduce boilerplate slightly and keep the directory a bit cleaner. Are there compelling reasons to keep them separate?

File: cypress/integration/visual-test/Button_spec.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+describe('Button visual regression check', () => {
```
Comment (dangerismycat) [inline cypress/integration/visual-test/Button_spec.js:1]: Given how similar these tests are, this seems like something we could script (possibly two scripts: one to generate test files for all our existing components, and another that's added to the new component scaffolding script)

File: docs/pages/visual-test/Button.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// @flow strict
+import type { Node } from 'react';
+import { Button } from 'gestalt';
+
+export default function ButtonSpec(): Node {
+  return <Button color="red" text="Save a Pin" />;
```
Comment (dangerismycat) [inline docs/pages/visual-test/Button.js:6]: How deep should we go with making sure we're testing every possible visual variation of a component? I would assume we'd want to test everything, but curious about your opinion

File: package.json (hunk 1 above)
Comment (dangerismycat) [reply package.json:102]: thank you! 😊

**Comment by dangerismycat** [comment]
This is super cool!

File: cypress/integration/visual-test/Button_spec.js (hunk 3 above)
Comment (christianvuerings) [reply cypress/integration/visual-test/Button_spec.js:1]: @dangerismycat agreed - that seems like a good follow up.

File: docs/pages/visual-test/Button.js (hunk 4 above)
Comment (christianvuerings) [reply docs/pages/visual-test/Button.js:6]: @dangerismycat we can go as deep as we want but I would start with adding a visual diff test for every component, only after we could add more tests.

**Review by dangerismycat — APPROVED** [review]
Approving as I don't think any of my questions should be blockers, and I'm interested to see this in action!

File: cypress/integration/visual-test/Button-dark_spec.js (hunk 2 above)
Comment (christianvuerings) [reply cypress/integration/visual-test/Button-dark_spec.js:1]: @dangerismycat I tried that but it doesn't work as expected - I always get back `cy.screenshot()` only works with a screenshot area with a height greater than zero.` on the second test, even when I change the assertions.

e.g.

```js
describe('Text visual regression check', () => {
  it('Compares screenshots: default', () => {
    cy.visit('/visual-test/Text');
    cy.get('[data-test-id="visual-test"]').should('contain', 'Text size small');
    cy.get('[data-test-id="visual-test"]').toMatchImageSnapshot({
      name: 'Text',
    });
  });

  it('Compares screenshots: dark mode', () => {
    cy.visit('/visual-test/Text-dark');
    cy.get('[data-test-id="visual-test"]').should('contain', 'Text size small (dark)');
    cy.get('[data-test-id="visual-test"]').toMatchImageSnapshot({
      name: 'Text-dark',
    });
  });
});
```

So going to leave this as is for now

Comment (dangerismycat) [reply cypress/integration/visual-test/Button-dark_spec.js:1]: Huh, weird. Well, not a big deal 🙂

**Pull request merged**